### DISCUSSION TITLE
OG-MARL Datasets are now hosted on Hugging Face :tada:

### DISCUSSION THREAD

**Discussion by jcformanek** [instadeepai/og-marl#31] [Announcements]
To address the challenges some users have faced trying to access OG-MARL datasets from certain global regions, we have moved the OG-MARL datasets to Hugging Face. We hope that this move will significantly improve the accessibility of OG-MARL datasets for the community.

:hugs: https://huggingface.co/datasets/InstaDeepAI/og-marl

**Comment by qiaodan-cuhk** [comment]
Hi Claude, I noticed that the datasets of some tasks in mamujoco like 2-ant, 4-ant, 3-hopper only include one dataset with quality "Replay" as the mixture of "Good", "Medium", and "Poor" on the latest huggingface resource. And these datasets links on the previous website https://sites.google.com/view/og-marl are no longer available. May I ask for the alternative methods to access the dataset "Good", "Medium", and "Poor"  of mamujoco 2-ant? Thanks! 😊

**Reply by jcformanek** [reply]
Hi there, I'm sorry about that. The GCP bucket that was hosting those datasets was unexpectedly deleted. But I think I have a backup somewhere. I'll get back to you as soon as possible and upload them to Hugging Face. The reason I have been putting it off is because the backup is in the old dataset format and I need to convert it to the new (vault) format. But Ill work on that now!

**Reply by qiaodan-cuhk** [reply]
Thank you for explaining the situation. I understand these things happen sometimes. I really appreciate your effort to recover and convert the backup datasets. Looking forward to the updated og-marl hugging-face repo! Thanks again for working on this. 🤗

**Reply by jcformanek** [reply]
After a little work, I got them converted. I have uploaded 2ant to hugging face. see here:

[https://huggingface.co/datasets/InstaDeepAI/og-marl/tree/main/core/mamujoco](https://huggingface.co/datasets/InstaDeepAI/og-marl/tree/main/core/mamujoco)

Ill work on 4ant too. I pushed a small change to og-marl so that it know where to look for the 2ant link! 

I did some quick tests and I think the dataset conversion was done correctly. But it is possible something went a bit wrong. So let me know if you pick up anything weird in the data. 

Hope this helps!

**Reply by qiaodan-cuhk** [reply]
Hi Claude, thank you so much for the quick update! It helps me a lot to evaluate my idea! I have checked the 2ant.vlt dataset with tutorial .ipynb files in ogmarl/examples and it works well. Here's are some output and I think they are quite similar to the other datasets.  Looking forward to the further updates about 4ant, etc. 🥳

**Reply by qiaodan-cuhk** [reply]
### Structure of Datasets 
`{'actions': (1, 1989488, 2, 4),
 'infos': {'state': (1, 1989488, 111)},
 'observations': (1, 1989488, 2, 54),
 'rewards': (1, 1989488, 2),
 'terminals': (1, 1989488, 2),
 'truncations': (1, 1989488, 2)}`.   

### Rewards
 `Reward at 0th step: [[-0.58427453 -0.58427453]]
Reward at 1th step: [[-2.923763 -2.923763]]
Reward at 2th step: [[-0.8254433 -0.8254433]]
Reward at 3th step: [[-0.37940472 -0.37940472]]
Reward at 4th step: [[0.69074035 0.69074035]]
Reward at 5th step: [[0.84611523 0.84611523]] `.  

### Single Timestep
 `{'actions': Array([[[-0.39739022, -0.9605442 ,  0.7134905 , -0.5490379 ],
         [ 0.61904144, -0.05751085, -0.06487702, -0.08025056]]],      dtype=float32),
 'infos': {'state': Array([[ 3.94978791e-01,  9.96950090e-01, -1.22380850e-03,
          -3.30485962e-02, -7.06877336e-02, -5.25079906e-01,
           5.06057560e-01, -5.40855408e-01, -5.21158814e-01,
           ...........
           0.52618796, -0.0115572 , -0.5250799 , -0.5408554 ,
           0.11637186,  0.09489634]]], dtype=float32),
 'rewards': Array([[0.65662473, 0.65662473]], dtype=float32),
 'terminals': Array([[0., 0.]], dtype=float32),
 'truncations': Array([[0., 0.]], dtype=float32)}`
